projects/Mallard/src/hooks/use-issue-summary.tsx

```diff
@@ -0,0 +1,110 @@
+import React, {
+    createContext,
+    useContext,
+    useState,
+    SetStateAction,
+    Dispatch,
+    useEffect,
+} from 'react'
+import { IssueSummary } from '../common'
+import { CachedOrPromise } from 'src/helpers/fetch/cached-or-promise'
+import { useCachedOrPromise } from './use-cached-or-promise'
+import { withResponse } from 'src/helpers/response'
+import { useNetInfo } from './use-net-info'
+import { fetchAndStoreIssueSummary, readIssueSummary } from '../helpers/files'
+import { PathToIssue } from 'src/paths'
+import NetInfo from '@react-native-community/netinfo'
+
+interface IssueSummaryState {
+    issueSummary: IssueSummary[] | null
+    issueId: PathToIssue | null
+    setIssueId: Dispatch<SetStateAction<PathToIssue | null>>
+    error: string
+}
+
+const IssueSummaryContext = createContext<IssueSummaryState>({
+    issueSummary: null,
+    issueId: null,
+    setIssueId: () => {},
+    error: '',
+})
+
+const getIssueSummary = (
+    isConnected = true,
+): CachedOrPromise<IssueSummary[]> => {
```

Comment:
By the looks of it we can completely ignore the `CachedOrPromise` abstraction here and just return a plain old `Promise`. I think the only code that ever needed wrapper was `useAPI`, which you've removed.